pgmpy/prediction/DynamicDMLRegressor.py

```diff
@@ -0,0 +1,851 @@
+#!/usr/bin/env python3
+"""
+Dynamic Double/Debiased Machine Learning for Sequential Treatment Effects.
+"""
+
+import warnings
+
+import numpy as np
+import pandas as pd
+from scipy.stats import norm
+from sklearn.base import BaseEstimator, RegressorMixin, clone
+from sklearn.ensemble import RandomForestRegressor
+from sklearn.linear_model import LinearRegression
+from sklearn.model_selection import KFold
+from sklearn.utils.validation import check_array, check_is_fitted
+
+
+class DynamicDMLRegressor(RegressorMixin, BaseEstimator):
+    """
+    Dynamic Double/Debiased Machine Learning for sequential treatment effects.
+    Estimates causal effects of treatments at different time periods on a
+    final outcome using recursive g-estimation with Neyman orthogonal moments.
+
+    Parameters
+    ----------
+    causal_graph : DAG, PDAG, ADMG, MAG, or PAG, optional
+        Causal graph with defined variable roles. If provided, the graph must have
+        the following roles: `exposure`, `outcome`. Additionally, `adjustment` and
+        `pretreatment` can be specified for confounders.
+        If None, all features in X are treated as confounders, and T/y column names
+        are used directly.
```

Comment:
Not sure how we distinguish `T` from `X` here? The fit method would only give the `X` and `y` dataframes, how do we extract `T` from `X`?